service/matching/physical_task_queue_manager.go

```diff
@@ -324,7 +324,7 @@ func (c *physicalTaskQueueManagerImpl) PollTask(
 	}
 
 	task.namespace = c.partitionMgr.ns.Name()
-	task.backlogCountHint = c.backlogMgr.BacklogCountHint
+	task.backlogCountHint = c.backlogMgr.db.getApproximateBacklogCount
```

Comment:
Hmm.. this value here is not "legacy api", this is important workflow progress api. It's used by SDKs to decide whether to poll sticky queues vs normal queue. If this is nonzero when there are no tasks immediately available on the sticky queue, then it will cause pollers to get stuck and delay workflow progress.

For worker scaling it's more important to be accurate at the higher end and not have an artificial limit, which ApproximateBacklogCount does, but for this purpose it's most important to correctly differentiate between 0 and > 0. How confident are we in that? Keep in mind that this will get called immediately after the task is acked, before anything is persisted.

Reply:
I think it is more likely for ApproximateBacklogCount to report non-0 when backlog is in fact empty, compared to the legacy BacklogCountHint. If that increases the chance of pollers getting stuck, then we should not use ApproximateBacklogCount in here.

In our offline discussion, @dnr suggested to simplify the BacklogCountHint logic to return only 0/1 (only in poll response, in legacy DescribeTaskQueue we'll report ApproximateBacklogCount). Need to confirm with SDK to make sure this behavior change does not break anything on their side.

Reply:
I agree. Changes will be made to restore this. Thank you!